### PR TITLE
Add GitHub action to push image to Quay

### PR DESCRIPTION
**Pull request by johnbieren**
Whenever a new commit is pushed to the main branch
of the repo, the bundle will be built and the result
will be pushed to quay.io/redhat-appstudio/release-service

Signed-off-by: Johnny Bieren <jbieren@redhat.com>

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+jobs:
+  build-image:
+    name: Build image
+    runs-on: ubuntu-20.04
+    env:
+      OPERATOR_SDK_VERSION: v1.18.0
+      QUAY_ORG: redhat-appstudio
+      QUAY_REPO: release-service
+    steps:
+      - name: Set up Go 1.x
```
Comment (davidmogar) [inline .github/workflows/push.yml:15]:
```suggestion
      - name: Set up Go
```

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        run: |
+          mkdir -p ~/bin
+          cp ~/cache/operator-sdk-${OPERATOR_SDK_VERSION} ~/bin/operator-sdk
+          echo "$HOME/bin" >> $GITHUB_PATH
+      - name: Regenerate executables with current environment packages
+        run: |
+          rm -f bin/kustomize
+          rm -f bin/controller-gen
+          make kustomize
+          make controller-gen
```
Comment (davidmogar) [inline .github/workflows/push.yml:43]: I guess they can go together (make kustomize controller-gen). Less lines. Same for rm commands.

**Review by davidmogar — COMMENTED** [review]
Very good. Just some minor comments.

**Comment by johnbieren** [comment]
This can't be merged until the PR with the Release CRD is added (because make bundle won't work as talked about on slack)

**Pull request merged**